### DISCUSSION TITLE
Best way to "dissolve" useless edges and vertex?

### DISCUSSION THREAD

**Discussion by FrankEscobar** [libigl/libigl#2185] [Q&A]
I mean when some faces are contiguous and share the same plane you can remove that edges (and some of their vertices) and merge that faces, without losing details since they are not relevant to the general detail.

Probably a threshold should be applied since you need to know what "share the same plane" means in terms of accuracy.

Here is a sample:
![image](https://user-images.githubusercontent.com/39060989/228981137-19dae8f8-f8c8-423e-8119-96cd12f2f5aa.png)

Any idea about this?

Thank you in advance.

**Comment by alecjacobson** [comment]
There're two things I'd try here:

1. `igl::copyleft::cgal::simplify_polyhedron` this will remove any unnecessary vertices in _perfectly flat_ regions of an input triangle mesh. The output will be a triangle mesh.

2. It looks like that image is using polygons (i.e., not just triangles). libigl doesn't have a lot of support for those, but you could still identify triangles that can be merged together by doing a breadth first search and clustering triangles according if their normals match. Since most downstream apps (and even file formats) only allow simple polygons (e.g., no annulus). Then you'd have to come up with a way to prevent those during the BFS or break them up as a post process.

**Comment by FrankEscobar** [comment]
Thank you, that image was taken from the internet, in my case, the source is a pure triangle mesh.

Is there any alternative that is not under copyleft? I was thinking about igl::collapse_edge if the edge vertices were using the same normal or something like that.

Thank you in advance!

**Comment by alecjacobson** [comment]
You could cook up the same thing as simplufy_polygon using doubles.
Wouldn't be exact but would often work fine.
